packages/pwa-buildpack/src/WebpackTools/PWADevServer.js

```diff
@@ -35,27 +35,28 @@ const helpText = `
 
 const PWADevServer = {
     validateConfig: optionsValidator('PWADevServer', {
-        publicPath: 'string',
-        env: 'object'
+        projectConfig: 'object'
     }),
     async configure(config) {
         debug('configure() invoked', config);
         PWADevServer.validateConfig('.configure(config)', config);
+
+        const { projectConfig } = config;
+
+        const devServerEnvVars = projectConfig.section('devServer');
+
         const devServerConfig = {
-            public: process.env.PWA_STUDIO_PUBLIC_PATH || '',
             contentBase: false, // UpwardPlugin serves static files
             compress: true,
             hot: true,
             watchOptions: {
                 // polling is CPU intensive - provide the option to turn it on if needed
                 poll:
-                    !!parseInt(
-                        process.env.PWA_STUDIO_HOT_RELOAD_WITH_POLLING
-                    ) || false
+                    !!parseInt(devServerEnvVars.watchOptionsUsePolling) || false
```

Comment:
Is the `|| false` is redundant here?

Reply:
Potentially, but that option may be required to be boolean. Webpack does that sometimes.